airbyte-commons-protocol/src/main/java/io/airbyte/commons/protocol/AirbyteMessageMigrator.java

```diff
@@ -25,104 +24,59 @@
 @Singleton
 public class AirbyteMessageMigrator {
 
-  private final List<AirbyteMessageMigration<?, ?>> migrationsToRegister;
-  private final SortedMap<String, AirbyteMessageMigration<?, ?>> migrations = new TreeMap<>();
-  private String mostRecentMajorVersion = "";
+  private final MigrationContainer<AirbyteMessageMigration<?, ?>> migrationContainer;
 
-  public AirbyteMessageMigrator(List<AirbyteMessageMigration<?, ?>> migrations) {
-    migrationsToRegister = migrations;
-  }
-
-  public AirbyteMessageMigrator() {
-    this(Collections.emptyList());
+  public AirbyteMessageMigrator(final List<AirbyteMessageMigration<?, ?>> migrations) {
+    migrationContainer = new MigrationContainer<>(migrations);
```

Comment:
Should the `MigrationContainer` be injected here instead of the `List<>`?

Reply:
I don't think so, the intent is to inject the list of migrations. We have different migrations intended for the `AirbyteMessageMigrator` or the `ConfiguredAirbyteCatalogMigrator`.
The `MigrationContainer<>` is a shared container for abstracting the logic of how we fetch the migrations. Injecting the MigrationContainer would involve an extra layer of subclassing which looks more like overhead than solving a problem at the moment.